### DISCUSSION TITLE
ControlNet Inpainting for SDXL

### DISCUSSION THREAD

**Discussion by ajkrish95** [Mikubill/sd-webui-controlnet#2225] [Ideas]
Currently we don't seem to have an ControlNet inpainting model for SD XL. Similar to the this - https://github.com/Mikubill/sd-webui-controlnet/discussions/1143 - are we planning to have a ControlNet Inpaint model?

**Comment by ghost** [comment]
Yes! I am deeply missing the Inpaint model when using XL. Hopefully we will see one. 🙏

**Reply by chlowden** [reply]
I too am looking for an inpaint SDXL model. Is there a particular reason why it does not seem to exist when other controlnets have been developed for SDXL? Or there a more modern technique that has replaced inpainting? Thank you

**Reply by JetsprintRacer** [reply]
Well... Fooocus, which is SDXL only WebUI, has built-in Inpainter, which works the same way as ControlNet Inpainting does with some bonus features.

**Reply by chlowden** [reply]
Thankyou for introducing me to Fooocus. It looks very interesting. But I'm looking for SDXL inpaint to upgrade a video comfyui  workflow that works in SD 1.5 at the moment.

**Comment by Lalimec** [comment]
any news on this?

**Reply by 2blackbar** [reply]
it workyes, use forge webui, its indentical to auto11 but has extra stuff and inpaint for SDXL , you have to take model from fooocus gui and copy it to forge

**Reply by Lalimec** [reply]
to where exactly, is this a controlnet model or an inpainting merge? And it is a .patch file.

**Reply by Lalimec** [reply]
found it, https://github.com/lllyasviel/stable-diffusion-webui-forge/commit/fb2e2716683453e8b0ba775f7596d8c9b080177d

**Reply by kennethjmp** [reply]
@2blackbar can you explain that?

**Reply by FruitPigFoot** [reply]
me too, i need some guide. help me out please

**Reply by dante28290** [reply]
download model inpaint_v26.fooocus.patch from this https://huggingface.co/lllyasviel/fooocus_inpaint/tree/main and copy to controlnet model folder and use it for sdxl

**Reply by 2blackbar** [reply]
Yes it does work in fooocus and in sd forge, it does NOT work in auto11 , just download the patch file  from here  https://huggingface.co/lllyasviel/fooocus_inpaint/tree/main               get  [inpaint_v26.fooocus.patch](https://huggingface.co/lllyasviel/fooocus_inpaint/blob/main/inpaint_v26.fooocus.patch)    from the link i pasted and use it like normal controlnet model, tested it and works in forge.
Would be nice to have it in auto11 as well but it does not work so far.

**Reply by LucianoDaluz** [reply]
How to download this

**Reply by LeF0URBE** [reply]
@LucianoDaluz https://huggingface.co/lllyasviel/fooocus_inpaint/tree/main 
like any other model on hugging face, there is a little download button on each model lines (in "files and version")

**Reply by vkleinmp** [reply]
Could you please elaborate how to use it in Forge? I've tried it as controllnet with inpainting, with and without inpaint preprocessor, tried all filling methods in inpaint.

**Reply by FruitPigFoot** [reply]
y'all tried controlnet inpaint with fooocus model and  canny  sdxl model at once? When i try using them both in txt2img, the result seems to show that it's not properly using inpaint mask that i gave(both when given mask image or drawing). Am i the only one experiencing this?

**Comment by roxmay** [comment]
First of all. Many thanks for the work !
I would also love a ControlNet Inpaint model ❤️

**Comment by zero41120** [comment]
The code commit on a1111 indicates that SDXL Inpainting is now supported. You can find more details here: [a1111 Code Commit](https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/14390).

**Comment by aniket-professional2025** [comment]
How to download the model path for controlnet that supports sdxl?

**Reply by LucianoDaluz** [reply]
https://huggingface.co/xinsir/controlnet-union-sdxl-1.0/tree/main

Use the PROMAX version for SDXL
